magda-correspondence-api/src/CContentApiDirMapper.ts

```diff
@@ -0,0 +1,92 @@
+import buildJwt from "@magda/typescript-common/dist/session/buildJwt";
+import * as rp from "request-promise-native";
+import * as mime from "mime-types";
+import * as recursiveReadDir from "recursive-readdir";
+import * as fse from "fs-extra";
+import * as path from "path";
+import * as typeis from "type-is";
+
+class CContentApiDirMapper {
```

Comment:
Why not use content API exclusively? Instead of maintaining a separate file system thing.

Reply:
Yes. It's fully rely on `content API`.
The `CContentApiDirMapper` is more for auto-deployment. 
It only uploads all local files in `emailTpls` to content API --- suppose new deployment or production site upgrade have no those resources in the beginning.
`CContentApiDirMapper` will make sure if any of those files missing, they will be auto - sync to `content API`.
The actual resource retrieving is all from `content API` directly.
It will also help future deployment.
Let's say we modify the default template and added a more image file.
The deployment doesn't require manual uploading the new file to the `content API` of the existing site.

Reply:
Yes. We can put it to `content DB migrator`. However, if we decide to put it to `content DB migrator`, we have to update it everything we have some changes. And it could including some images files (think about future UI changes).
Right there is no extra work needs to be done and all new files will be auto-included. Was trying to avoid some maintenance troubles (for whoever has to work on it) 😄